# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1318043078 14400
#      Fri Oct 07 23:04:38 2011 -0400
# Node ID 007ebc128ab5ffbc7f9ba155fa2fb93088a9c8d9
# Parent  10eba5cdb5d4f6e75eb317465c698d58c1bf3998
fix input validation for sortrows

* sortrows.m: Don't check values of second argument unless
nargin is 2.

diff --git a/scripts/general/sortrows.m b/scripts/general/sortrows.m
--- a/scripts/general/sortrows.m
+++ b/scripts/general/sortrows.m
@@ -32,20 +32,22 @@
 ## Adapted-by: jwe
 
 function [s, i] = sortrows (A, c)
 
   if (nargin < 1 || nargin > 2)
     print_usage ();
   endif
 
-  if (! (isnumeric (c) && isvector (c))) 
-    error ("sortrows: C must be a numeric vector");
-  elseif (any (c == 0) || any (abs (c) > columns (A)))
-    error ("sortrows: all elements of C must be in the range [1, columns (A)]");
+  if (nargin == 2)
+    if (! (isnumeric (c) && isvector (c))) 
+      error ("sortrows: C must be a numeric vector");
+    elseif (any (c == 0) || any (abs (c) > columns (A)))
+      error ("sortrows: all elements of C must be in the range [1, columns (A)]");
+    endif
   endif
 
   default_mode = "ascend";
   reverse_mode = "descend";
 
   if (issparse (A))
     ## FIXME: Eliminate this case once __sort_rows_idx__ is fixed to
     ##        handle sparse matrices.
